rich/bar.py

```diff
@@ -195,6 +195,118 @@ def __rich_measure__(self, console: Console, max_width: int) -> Measurement:
         )
 
 
+BLOCK_ELEMS = [" ", "▏", "▎", "▍", "▌", "▋", "▊", "▉"]
+FULL_BLOCK = "█"
+
+
+class BlockBar(JupyterMixin):
+    """Renders a (progress) bar.
+
+    Args:
+        total (float, optional): Number of steps in the bar. Defaults to 100.
+        completed (float, optional): Number of steps completed. Defaults to 0.
+        width (int, optional): Width of the bar, or ``None`` for maximum width. Defaults to None.
+        style (StyleType, optional): Style for the bar background. Defaults to "bar.back".
+        complete_style (StyleType, optional): Style for the completed bar. Defaults to "bar.complete".
+        finished_style (StyleType, optional): Style for a finished bar. Defaults to "bar.done".
+        blend_colors (bool, optional): Blend colors according to completion. Defaults to False.
+    """
+
+    def __init__(
+        self,
+        total: float = 100,
+        completed: float = 0,
+        width: int = None,
+        style: StyleType = "bar.back",
+        complete_style: StyleType = "bar.complete",
+        finished_style: StyleType = "bar.finished",
+        blend_colors: bool = False,
```

Comment:
I think that we probably don't want to implement blending here. This class could be a primitive that you could extend to add the blend functionality.

It should probably use colors, rather than styles. Since much of the attributes (italic / bold) etc don't make sense for a solid bar.

Here's what I'm thinking for the constructor:

```
width: Optional[float],  # Number of cells to use, or None for full width
size: float, # Value for end of the bar
start: float,  # Start point (in 0 -> size units)
end: float,  # End point
color: Union[Color, str] = "default",
background: Union[Color, str] = "default", 
```
A union for the colors, so you can pass a Color or a string to call Color.parse with.

What do you think?